core/src/providers/mistral.rs

```diff
@@ -394,8 +400,21 @@ impl MistralAILLM {
         Ok(mistral_messages)
     }
 
-    fn tokenizer(&self) -> Arc<RwLock<CoreBPE>> {
-        return p50k_base_singleton();
+    fn tokenizer(&self) -> Arc<RwLock<SentencePieceProcessor>> {
+        if self.id.starts_with("mistral-tiny")
+            || self.id.starts_with("mistral-embed")
+            || self.id.starts_with("open-mistral-7b")
```

Comment:
Where is the mapping between those id and the one we display?

Reply:
I don't think it's a mapping per se, but it looks like we are missing some of the available models in `SupportedModels`. Will look into how this can be fixed.

Reply:
Ok cause I have not seen any of the `open-*` models in our codebase.

Reply:
Yep I think originally those were not in the API, but from the docs they seem to be now. And I'm curious to test the new MoE 8x22b